pykafka/protocol.py

```diff
@@ -125,6 +127,45 @@ def raise_error(self, err_code, response):
                 clsname, response))
 
 
+class MessageOffset(object):
+    """A message offset described as a pair:
+    the offset of the message or its containing compressed message,
+    and the offset of the submessage in its containing message, if any."""
+    __slots__ = ("main", "sub")
+
+    def __init__(self, main_offset, sub_offset=None):
+        self.main = main_offset
+        self.sub = sub_offset
+
+    def can_precede(self, other, compressed_topic):
+        """Can those two pairs follow one another?"""
+        if other.sub is None:
+            return (self.main + 1 == other.main) or (
+                compressed_topic and (self.main + 1 < other.main))
+        else:
+            # Compressed message offset may not follow immediately
```

Comment:
This comment doesn't seem to directly apply to the code that follows it. The way I'm explaining this code to myself is

> Two messages, both from compressed message sets, can be sequential in one of exactly two different ways: either they came from the same message set (`self.main == other.main`) and have sequential sub-offsets, or `self` came from the compressed message set directly preceding that of `other`, and `other` is the first message in its set.